plugins/fda_drugs/dumper.py

```diff
@@ -56,102 +74,224 @@ def post_dump(self, *args, **kwargs):
         with zipfile.ZipFile(local_zip_file, "r") as zip_object:
             zip_object.extractall(data_directory)
 
+        unique_entries = set()
+        products_content = self._process_product_file(data_directory=data_directory, unique_entries=unique_entries)
+        marketing_status_content = self._process_marketing_status_file(
+            data_directory=data_directory, unique_entries=unique_entries
+        )
+        te_content = self._process_te_file(data_directory=data_directory, unique_entries=unique_entries)
+        applications_content = self._process_applications_file(data_directory=data_directory)
+
         marketing_status_lookup_file = Path(data_directory).joinpath("MarketingStatus_Lookup.txt")
         marketing_status_lookup_table = self._build_marketing_status_lookup_table(marketing_status_lookup_file)
 
-        products_file = Path(data_directory).joinpath("Products.txt")
-        marketing_status_file = Path(data_directory).joinpath("MarketingStatus.txt")
-        te_file = Path(data_directory).joinpath("TE.txt")
+        group_file_path = Path(data_directory).joinpath(self.grouped_file)
+        join_mapping = []
+        for lookup_key in unique_entries:
+            product = products_content.get(lookup_key, NULL_PRODUCT)
+            marketing_status = marketing_status_content.get(lookup_key, NULL_MARKETING_STATUS)
+            te = te_content.get(lookup_key, NULL_TE)
+
+            marketing_status_value = marketing_status_lookup_table.get(marketing_status.MarketingStatusID, None)
+
+            application_numbers = [product.ApplNo, marketing_status.ApplNo, te.ApplNo]
+            application_number = [appl for appl in application_numbers if appl is not None][0]
+
+            product_numbers = set([product.ProductNo, marketing_status.ProductNo, te.ProductNo])
+            product_number = [prod for prod in product_numbers if prod is not None][0]
+
+            application = applications_content.get(application_number, NULL_APPLICATION)
+            application_company = None
+            if application.ApplPublicNotes is not None:
+                application_company = application.ApplPublicNotes
+            elif application.SponsorName is not None:
+                application_company = application.SponsorName
+
+            unique_id = f"ApplNo-{application_number}-ProductNo-{product_number}"
```

Comment:
@ctrl-schaff I think we can simplify this `_id` field as `f"{application_number}-{product_number}"`? We have both values in the object, so it's easy to tell what they are.

Reply:
That's fine by me